examples/src/main/java/com/mapbox/navigation/examples/core/SilentWaypointsRerouteActivity.kt

```diff
@@ -0,0 +1,377 @@
+            registerRoutesObserver(routeObserver)
+            registerRouteProgressObserver(routeProgressObserver)
+            registerOffRouteObserver(this@SilentWaypointsRerouteActivity)
+            getRerouteController()?.registerRerouteStateObserver(
+                this@SilentWaypointsRerouteActivity
+            )
+        }
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        mapView.onResume()
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        mapView.onPause()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapboxNavigation?.run {
+            unregisterLocationObserver(locationObserver)
+            unregisterTripSessionStateObserver(tripSessionStateObserver)
+            unregisterRouteProgressObserver(routeProgressObserver)
+            unregisterRoutesObserver(routeObserver)
+            unregisterOffRouteObserver(this@SilentWaypointsRerouteActivity)
+            getRerouteController()?.unregisterRerouteStateObserver(
+                this@SilentWaypointsRerouteActivity
+            )
+        }
+        mapView.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapboxReplayer.finish()
+        mapboxNavigation?.stopTripSession()
+        mapboxNavigation?.onDestroy()
+        mapView.onDestroy()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView.onLowMemory()
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap, this, true)
+
+            // Center the map at current location. Using LocationEngineProvider because the
+            // replay engine won't have your last location.
+            LocationEngineProvider
+                .getBestLocationEngine(this)
+                .getLastLocation(locationListenerCallback)
+
+            directionRoute?.let {
+                navigationMapboxMap?.drawRoute(it)
+                mapboxNavigation?.setRoutes(listOf(it))
+                btnStartNavigation.visibility = VISIBLE
+            }
+        }
+
+        mapboxMap.addOnMapLongClickListener { latLng ->
+            waypointsController.add(latLng)
+            routeSettings.destination = latLng.toPoint()
+            mapboxMap.locationComponent.lastKnownLocation?.let { originLocation ->
+                mapboxNavigation?.requestRoutes(
+                    RouteOptions.builder().applyDefaultParams()
+                        .accessToken(Utils.getMapboxAccessToken(applicationContext))
+                        .coordinates(waypointsController.coordinates(originLocation))
+                        .waypointIndices("0;${waypointsController.waypoints.size}")
+                        .alternatives(true)
```

Comment:
The example does not handle alternative routes, let's just set this to `false`.

Reply:
We can even remove it as by default is set to `false` 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/356e9d150b2e1ca5ea5b9c7222bc222316d60d23/libnavigation-router/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt#L34

Reply:
👍